### PR TITLE
Toolchain/Ports/cmake: Update cmake to version 3.26.4

### PR DESCRIPTION
**Pull request by shannonbooth**

### CODE REVIEW

**Comment by timschumi** [comment]
This should probably also update all other CMake versions across the board (Toolchain, etc.). Also, doesn't this require the linked patch to build for SerenityOS then?

**Comment by shannonbooth** [comment]
> This should probably also update all other CMake versions across the board (Toolchain, etc.). Also, doesn't this require the linked patch to build for SerenityOS then?

Ah, didnt think of updating across the board, will do.

The linked patch is just a compiler warning, it builds and works fine without it.

**Comment by shannonbooth** [comment]
Bumped the version in toolchain as well!

**Comment by timschumi** [comment]
> The linked patch is just a compiler warning, it builds and works fine without it.

I'm pretty sure that it doesn't on Clang.

Anyways, not like it will be broken by this particular version bump, it already is broken ever since the Clang update. Feel free to add or not add the patch.

**Comment by shannonbooth** [comment]
> 



> > The linked patch is just a compiler warning, it builds and works fine without it.
> 
> I'm pretty sure that it doesn't on Clang.
> 
> Anyways, not like it will be broken by this particular version bump, it already is broken ever since the Clang update. Feel free to add or not add the patch.

Ah interesting, yeah I didn't bother patching it at the time since I figured it would have always been broken considering that bit of code in kwsys has been around for many years. Will add the patch, thanks!

File: Meta/CMake/cmake-version.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Note: Update this alongside Toolchain/BuildCMake.sh
 set(version_ok 0)
-if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.25.1)
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.26.4)
```
Comment (timschumi) [inline Meta/CMake/cmake-version.cmake:3]: As a side note, do we have to force users to rebuild CMake? As long as there aren't any new required patches that have landed upstream we probably shouldn't.

Comment (shannonbooth) [reply Meta/CMake/cmake-version.cmake:3]: Yeah, I was considering this too, but the comment in that file said to update here at same time, so I just followed it 🤷

But yeah, you're right, no point. I'll update the comment to along those lines

**Review by ADKaster — APPROVED** [review]
This seems fine as-is. The point of the check was to make sure that if we upstreamed some required changes to CMake, we would keep the min version check in sync with the version we're building. But as we haven't had a need to upstream any cmake recently, this is fine the way you've updated it.

**Pull request merged**